R/logistics_helper.R

```diff
@@ -0,0 +1,72 @@
+## log-likelihood
+logit_loglik <- function(beta, X, Y){
+  p <- 1 / (1 + exp(-X %*% beta))
+  l <- -(t(Y) %*% log(p) + t(1 - Y) %*% log(1 - p))
+  return(l)
+}
+
+
+## gradient
+logit_gradient <- function(beta, X, Y){
+  y_est <- 1 / (1 + exp(-X %*% beta))
+  gr = as.vector(t(X) %*% (y_est - Y))
+  return(gr)
+}
+
+## BFGS
+BFGS_finder_logit <- function(design, outcome, method){
```

Comment:
-2

This function is almost identical to your linear regression version. Better to write a single `bfgs()` function that takes in design, outcome, function, and gradient. Then you can pass in the desired likelikhood and gradient.